elastic-agent-client.proto

```diff
@@ -323,10 +323,12 @@ message UnitObserved {
 message CheckinObservedVersionInfo {
     // Name of the binary.
     string name = 1;
-    // Version of the binary.
+    // Agent package version sent by the agent.
```

Comment:
Thinking a bit more, if this field always needs to equal the agent version there's no value in reporting it. Should we just remove this field entirely?

If we want to keep it, we can have the client code here automatically populate it.

Reply:
Yes, it does not provide any value. I'd argue it should be still the binary real version, but this train already left. 
However, keeping the current behaviour does not harm, do not require any change and would allow us in the future to use somehow the real binary version if we ever want. Therefore I'd argue doing nothing is the best option.

Reply:
Best to keep it for backwards compatibility.